### ISSUE TITLE
ParserJS: validate document only with the AsyncAPI v3 JSON Schema

### ISSUE THREAD

**Issue by Amzani** [asyncapi/parser-js#817]
#### Reason/Context

We need a working version of ParseJS that we can use both in the CLI and Studio to validate Async 3.0 documents.

```
import { Parser } from '@asyncapi/parser' // using 2.1.0-next-major-spec.1

const parser = new Parser()

const documentRaw = {
    asyncapi: '3.0.0',
    info: {
      title: 'Invalid AsyncApi document',
      version: '1.0',
    }
}

const { document, diagnostics } = await parser.parse(documentRaw)
console.log(document, diagnostics)

```

Current output
```
=> current output
undefined [
  {
    code: 'asyncapi-is-asyncapi',
    message: 'Version "3.0.0" is not supported. Please use "2.6.0" (latest) version of the specification.',
    path: [],
    severity: 0,
    range: { start: [Object], end: [Object] }
  }
] 
```

#### What triggered this issue 

https://asyncapi.slack.com/archives/C0230UAM6R3/p1690292367285459

#### Out of scope

- Spectral rules are out of scope, we only focus on the validation

#### Solution

(From @magicmatatjahu, some more details might be missing)

- Format should be added here https://github.com/asyncapi/parser-js/blob/master/src/ruleset/formats.ts to support 3.0.0
- Bump the json-spec-schemas to the latest version (with 3.0.0 - even rc - schema included) -> https://github.com/asyncapi/parser-js/blob/master/src/ruleset/functions/isAsyncAPIDocument.ts 
- Pump @asyncapi/specs dependency that is currently using `5.1.0`

Related to https://github.com/asyncapi/studio/issues/742

**Comment by smoya** [comment]
/progress 70 Mainly working, but blocked by https://github.com/asyncapi/studio/issues/742

**Comment by smoya** [comment]
/progress 85 Working on the following fix https://github.com/asyncapi/parser-js/issues/838

**Comment by smoya** [comment]
/progress 90 All the features and bugfixes are ready in `next-major-spec`
